internal/k8s_watcher/watcher.go

```diff
@@ -194,21 +178,18 @@ func (c *Watcher) watchMetamorph() {
 						c.logger.Info("records unlocked", slog.Int64("rows-affected", resp), slog.String("pod-name", podName))
 					}
 				}
-
 				runningPods = runningPodsK8s
-
-			case <-c.shutdownMetamorph:
-				return
 			}
 		}
 	}()
 }
 
 func (c *Watcher) Shutdown() {
-	c.tickerMetamorph.Stop()
-	c.shutdownMetamorph <- struct{}{}
-	<-c.shutdownMetamorphComplete
```

Comment:
If we don't wait for a signal when the go routine is finished, then I think the shutdown is not graceful anymore i.e. the logic from line 172 to 196 may be interrupted, not?

It could also be done with a wait group instead of a channel